hoomd/mpcd/RadialSolventVelocityAnalyzer.cc

```diff
@@ -0,0 +1,201 @@
+    {
+    if (m_prof) m_prof->push("RadialSolventVelocityAnalayse");
+
+    binParticles();
+
+    accumulate();
+
+    if (m_prof) m_prof->pop();
+    }
+
+/*!
+ * Particle pairs are binned into *m_counts* 
+ */
+void RadialSolventVelocityAnalyzer::binParticles()
+    {
+    if (m_prof) m_prof->push("bin");
+
+    ArrayHandle<Scalar4> h_pos(m_mpcd_pdata->getPositions(), access_location::host, access_mode::read);
+    ArrayHandle<unsigned int> h_counts(m_counts, access_location::host, access_mode::overwrite);
+    memset(h_counts.data, 0, m_num_bins * sizeof(unsigned int));
+
+    const Scalar Rsq = m_Rmax * m_Rmax;
+
+    // compute the total number of mpcd particles
+    const unsigned int N = m_mpcd_pdata->getN();
+    for (unsigned int i=0; i < N; ++i)
+        {
+        // load particle i
+        const Scalar4 pos_i = h_pos.data[i];
+        const Scalar3 r_i = make_scalar3(pos_i.x, pos_i.y, pos_i.z);
+
+        // distance calculation
+        const Scalar r_isq = dot(r_i, r_i);
+
+        if (r_isq < Rsq)
+            {
+            const unsigned int bin = static_cast<unsigned int>(slow::sqrt(r_isq) / m_bin_width);
+            ++h_counts.data[bin];
+            }
+        }
+
+    if (m_prof) m_prof->pop();
+    }
+
+/*!
+ * To get rho(r), we calculate as
+ *
+ * \verbatim
+ *                  counts_i
+ *      g_i = ----------------------
+ *                  V_shell
+ *
+ * \endverbatim
+ *
+ * The shell volume is V_shell = (4 pi/3)*(r_{i+1}^3-r_i^3).
+ */
+void RadialSolventVelocityAnalyzer::accumulate()
+    {
+    if (m_prof) m_prof->push("accumulate");
+
+    if (m_num_samples > 0)
+        {
+        m_accum_rho.resize(m_num_samples, m_num_bins);
+        }
+
+    ArrayHandle<unsigned int> h_counts(m_counts, access_location::host, access_mode::read);
+    ArrayHandle<double> h_accum_rho(m_accum_rho, access_location::host, access_mode::overwrite);
+
+    for (unsigned int i=0; i < m_num_bins; ++i)
+        {
+        const double r_in = m_bin_width * static_cast<double>(i);
+        const double r_out = std::min(r_in + m_bin_width, static_cast<double>(m_Rmax));
+        const double V_shell = (4.0*M_PI/3.0) * (r_out * r_out * r_out - r_in * r_in * r_in);
+        h_accum_rho.data[i + m_num_samples] += static_cast<double>(h_counts.data[i])/ V_shell;
+        }
+    ++m_num_samples;
+    if (m_prof) m_prof->pop();
+    }
+
+/*!
+ * \returns Bins distribution function was computed on
+ */
+std::vector<Scalar> RadialSolventVelocityAnalyzer::getBins() const
+    {
+    std::vector<Scalar> bins(m_num_bins);
+    for (unsigned int i=0; i < m_num_bins; ++i)
+        {
+        const Scalar r_in = m_bin_width * static_cast<Scalar>(i);
+        const Scalar r_out = std::min(r_in + m_bin_width, m_Rmax);
+        bins[i] = Scalar(0.5) * (r_in + r_out);
+        }
+    return bins;
+    }
+
+/*!
+ * \returns Accumulated density
+ */
+std::vector<std::vector<Scalar>> RadialSolventVelocityAnalyzer::get() const
```

Comment:
This will not do what you think it does because of your indexing. I recommend you make the signature of this method be:
```
std::vector<Scalar> RadialSolventVelocityAnalyzer::get(unsigned int idx) const
```
and have it return the profile that corresponds to the requested index.

At the Python level, you can then manipulate these vectors into NumPy arrays that have the shape / indexing we want.